src/node/context.h

```diff
@@ -70,6 +71,7 @@ struct NodeContext {
     std::unique_ptr<CBlockPolicyEstimator> fee_estimator;
     std::unique_ptr<PeerManager> peerman;
     std::unique_ptr<ChainstateManager> chainman;
+    std::shared_ptr<BlockTemplateCache> block_template_cache;
```

Comment:
In commit "node: add block template cache to miner" (4de7887ca47197654f480471db04750d44e405de)

I think it would probably be better for this to use unique_ptr instead shared_ptr (like fee_estimator and other members), so lifetime of the BlockTemplateCache is more explicit and predictable.